src/components/shipModal.html

```diff
@@ -9,24 +9,13 @@ <h5 class="modal-title">Regional Travel</h5>
             </div>
             <div class="modal-body bg-ocean">
                 <img src='assets/images/ship.png' style='width: 100%;'>
-                <div style='margin-top:20px'>
+                <div style='margin-top:20px' data-bind="foreach: GameHelper.enumStrings(GameConstants.Region)">
                     <button class='btn btn-block btn-success'
                         data-dismiss='modal'
-                        data-bind='visible: player.region != 0'
-                        onclick='MapHelper.moveToTown("Vermillion City");player.region = 0;'>
-                        Kanto
-                    </button>
-                    <button class='btn btn-block btn-success'
-                        data-dismiss='modal'
-                        data-bind='visible: player.region != 1 && player.highestRegion() >= 1'
-                        onclick='MapHelper.moveToTown("Olivine City");player.region = 1;'>
-                        Johto
-                    </button>
-                    <button class='btn btn-block btn-success'
-                        data-dismiss='modal'
-                        data-bind='visible: player.region != 2 && player.highestRegion() >= 2'
-                        onclick='MapHelper.moveToTown("Slateport City");player.region = 2;'>
-                        Hoenn
+                        data-bind='visible: player.region != $index && $index <= GameConstants.MAX_AVAILABLE_REGION,
```

Comment:
Shouldn't this be checking for `player.highestRegion()` somewhere?

Reply:
Ah, yeah good catch, fixed 👍